keps/sig-api-machinery/3962-mutating-admission-policies/README.md

```diff
@@ -0,0 +1,1383 @@
+  - Feature gate name: MutatingAdmissionPolicy
+  - Components depending on the feature gate: kube-apiserver
+
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+No, default behavior is the same.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+Yes, disabling the feature will result in mutation expressions being ignored.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+The MutatingAdmissionPolicy will be enforced again.
+
+###### Are there any tests for feature enablement/disablement?
+Unit test and integration test will be introduced in alpha implementation.
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+The existing workload could potentially be mutated and cause unexpected data stored if the mutation is misconfigured.
+
+While rollout, the cluster administrator could configure the feature with validation configured accordingly. In this way it will minimize the effect on the running workloads.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
```

Comment:
[non-blocking, this section is required for beta only]

We should probably discuss signals for clusters that _are_ opted in, to evaluate if they should rollback?

For example, you might want to rollback if you see elevated API server errors? 
Or maybe excessive `apiserver_cel_evaluation_duration_seconds` / `apiserver_cel_compilation_duration_seconds` ?

Reply:
Updated :)